src/string_enums/attack_result.hpp

```diff
@@ -0,0 +1,31 @@
+/*
+	Copyright (C) 2008 - 2021
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include <array>
+
+#include "string_enums/enum_base.hpp"
+
+struct attack_result_defines
+{
+	static constexpr const char* const hit = "hit";
+	static constexpr const char* const miss = "miss";
+	static constexpr const char* const kill = "kill";
+	static constexpr const char* const invalid = "invalid";
+
+	enum class type { HIT, MISS, KILL, INVALID, ENUM_MAX };
+	static constexpr std::array<const char*, static_cast<int>(type::ENUM_MAX)> values{ hit, miss, kill, invalid };
+};
+using attack_result = string_enums::enum_base<attack_result_defines>;
```

Comment:
While `attack_result` is more descriptive than `hit_type`, there's already a class with the same name in src/ai/actions.hpp. The other one gets forward-declared in src/ai/game_info.hpp,

Reply:
I can change it to `attack_results` unless there another name you would prefer?

Reply:
The AI one is the result of an attack, but this one is the result of a single blow within an attack. I'm not sure if that helps in naming it though…

The only other possible name I can think of right now is `attack_hit_type`.

Reply:
Suggestions: `swing_result` or `swing_type`. The animation code is already using `int swing_num` in this context.

Reply:
Maybe `strike_result`? I guess `swing_result` would tend to imply certain weapons (you don't swing things like a bow or a lance), but honestly I also don't really have a strong opinion any particular way.